tools/c7n_azure/c7n_azure/actions/base.py

```diff
@@ -64,10 +65,19 @@ def _process_resources(self, resources, event):
             try:
                 self._process_resource(r)
             except Exception as e:
-                self.log.error("Failed to process resource.\n"
-                               "Type: {0}.\n"
-                               "Name: {1}.\n"
-                               "Error: {2}".format(r['type'], r['name'], e))
+                if "pytest" in sys.modules:
```

Comment:
Ideally, I prefer to keep the same behavior for tests and real runs.. We can think about triggering some event to make tests easier but keep the behavior part.

If you think we have to do this, I'd prefer to use `C7N_TEST_RUN` env instead of looking for a specific module.

Reply:
I agree behavior should be the same, but I also worried that actions can fail and you get a passed test with no logs. 

I guess we could try to do something in our test base to watch for error logs and default to asserting on them.